README.md

```diff
@@ -4,13 +4,119 @@ Exercism problems in Ceylon.
+
+For the Ceylon track, the files needed to port an exercise are described in the next section about [exercise structure](#exercise-structure).
+
+### Exercise structure
+
+Each exercise is located in the `exercises` directory, in a directory with the same name as its slug (the `leap` exercise will be at `exercises/leap`, etc.).
+This directory should contain the following files and directories, where `<slug>` is the slug of the exercise, and `<Slug>` is the CamelCased version:
+
+* `source/<slug>/module.ceylon`: the [module descriptor](https://ceylon-lang.org/documentation/tour/modules/#dependencies_and_module_descriptors).
+  * This file will declare the module and any dependencies.
+  * Typically, the only dependency will be on [`ceylon.test`](https://herd.ceylon-lang.org/modules/ceylon.test) at the current Ceylon version.
+  * We don't expect to have other dependencies, but we may consider adding dependencies for an exercise if it would be unnecessarily difficult and/or cumbersome without those particular dependencies.
+* `source/<slug>/<Slug>.ceylon`: the stub solution.
+  * This file provides just the type signatures needed to make the tests compile.
+  * The implementations for all functions should all be `return nothing;`, which will intentionally fail the tests.
+* `source/<slug>/<Slug>Test.ceylon`: The test file.
+  * If there is an `exercises/<slug>/canonical-data.json` in the `x-common` repository, the test cases should typically be those described in that file.
+  * If you believe a case should be added or removed, try discussing it in the `x-common` repository to see if it would apply to all tracks.
+  * If a modification to the cases only makes sense for the Ceylon track, please note the change in the test file.
+  * If no JSON file is present, you may use the same tests as other languages implementing that exercise.
+  * Typically, we like to use [parameterized tests](https://ceylon-lang.org/blog/2016/02/22/ceylon-test-new-and-noteworthy/#parameterized_tests) to express the tests with minimal duplication in test code.
+* `example/<Slug>.ceylon`: An example solution that passes the tests.
+  * This will not be served to students (all paths containing `example` are not served).
+  * Instead, it is used in our [Travis CI run](.travis.yml) to make sure that the exercise is solvable and that the tests make sense.
+  * Other files may be included in the `example/` directory if appropriate.
+
+### Style and conventions
+
+Various choices were made in the past to get the repository to the state it is in now.
+These choices aren't set in stone; they could be changed at a later date.
+But the rationale behind the current choices made will be noted here.
+
+#### Style
+
+At the time of writing, the current maintainer of the Ceylon track was not able to find any commonly-accepted style guide for Ceylon.
+Nor did the maintainer find automatic linter or formatter.
```

Comment:
the maintainer did not find one at the time of writing, but the maintainer has since found that https://github.com/ceylon/ceylon.formatter exists.

Further, if https://github.com/ceylon/ceylon.formatter/commit/d583983da73b926a9a3f29b3624dbbe8d15975b9#diff-2ca0389ed527e942da6cc337da59153f is to be believed, this has existed since 1.1.0 and the maintainer took on the track at 1.3.1, so obviously the maintainer did not look hard enough!!